lib/plugins/aws/package/lib/packageService.js

```diff
@@ -54,6 +57,8 @@ module.exports = {
     const include = this.getIncludes();
     const zipFileName = this.provider.naming.getServiceArtifactName();
 
+    _.set(this.serverless.service, 'package.artifact', zipFileName);
```

Comment:
I think this is where things fall apart.

Changing it to:

```javascript
const foo = path.join(this.serverless.config.servicePath, '.serverless', zipFileName);

_.set(this.serverless.service, 'package.artifact', foo);
```

fixes the problem.